### ISSUE TITLE
Replay bot bugging out 

### ISSUE THREAD

**Issue by NikolaiVsevolodovich** [nikooo777/ckSurf#73]
Certain surf maps have the "golden knife," aka a map trigger which sets player_speedmod to a value higher than 1. It appears this speed multiplier is not reflected in the speed cksurf displays, nor is it reflected in cl_showpos 1. I'm trying to actually figure out how exactly this even modifies the player within the game engine because as far as I can tell so far no cvars are edited, nothing to do with speed, velocity, etc.. All I can think is speedmod is a hardcoded multiplier that occurs after all the in game calculations and relevant data is pulled.

Not sure if anyone has decided to try to fix this, but the server I'd like to fix it on I noticed is still running 1.18 so not sure if this has ever been addressed by any of you?

Bout to start digging myself. Thanks guys!

**Comment by fluffyst** [comment]
Easiest way to fix is to apply the speed modifier on the bot, the "golden knife" it self isn't setting the player_speedmod, there's a trigger that sets it on the same origin as where equipping the knife is done. 

I believe it is just setting the clients `m_flLaggedMovementValue` so perhaps you could save that value in the replay header and get that value and apply to replay bot when needed, but tbh I think just removing the player_speedmod via stripper is the best way to go about it.

`SetEntPropFloat(client, Prop_Data, "m_flLaggedMovementValue", 1.5);` will most likely be the same, or whatever the value for the player_speedmod is.

**Comment by NikolaiVsevolodovich** [comment]
Yeah I realize its not actually the weapon itself, but a trigger where the weapon is placed. Sounds like you know this inside and out haha. A kind soul on reddit pointed me to a section in player.cpp which handles the movement speed change on line 7892 as follows:
>  pPlayer->SetLaggedMovementValue( data.value.Float() );

So I believe you're right. I haven't tracked down SetLaggedMovementValue but I suppose its logical it does what you said.

Sorry for my ignorance, but what exactly do you mean by removing the player_speedmod via "stripper" ?

Realistically it's for a 24/7 map, and all the top times are going to be using the "knife" so hard coding it into the bot replay wouldn't make that big a deal. However I do prefer modularity.. I'd also like to apply the mutliplier to the displayed on screen speed. Another thing I noticed (and perhaps patched since 1.18) is that it will display an increased speed upto 300 or more when prestrafing, however cl_showpos remains locked at 250.

Sorry if I sound like a total dolt. I guess I should just get your latest version going and do my best to learn the ins and outs before I start deciding on more things I'd like to change haha.

**Comment by fluffyst** [comment]
[Stripper](https://forums.alliedmods.net/showthread.php?t=39439) is a useful metamod plugin which allows you to manipulate entities on any map without having to decompile/recompile, so you could easily remove the entity or trigger via Stripper and just delete times that used the exploit.

I believe since prestrafe changes a different value (maybe `m_flVelocityModifier`) rather than `m_flSpeed` which is probably what `cl_showpos 1` displays, since any prestafe in CS:GO is not "real" prestrafe, but man-made via a Plugin.
